src/shared/data-fd-util.h

```diff
@@ -5,3 +5,4 @@
 
 int acquire_data_fd(const void *data, size_t size, unsigned flags);
 int copy_data_fd(int fd);
+int memfd_clone_fd(int fd, const char *name, int mode);
```

Comment:
I'd prefer the addition of memfd_clone_fd() to be a separate commit, even if it doesn't have a test. The function is large enough to deserve to be its own thing.